File: back-end/models/Genre.go
package models

import (
	"time"

	"github.com/go-playground/validator/v10"
)

type Genre struct {
	IdGenre   uint      `gorm:"primaryKey" json:"id_genre"`
	NameGenre string    `json:"name_genre"`
	IsActive  bool      `json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (genre *Genre) Validate() error {
	validate := validator.New()
	return validate.Struct(genre)
}

File: front-end/web/src/app/models/genre.ts
export class Genre {
    id_genre: number;
    name_genre: string;
    is_active: boolean;
    created_at: Date;
    updated_at: Date;
  
    constructor() {
      this.id_genre = 0;
      this.name_genre = '';
      this.is_active = true;
      this.created_at = new Date();
      this.updated_at = new Date();
    }
  }
  